### PR TITLE
updates init method to produce static Textile

### PR DESCRIPTION
**Pull request by andrewxhill**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -29,56 +29,62 @@ export * from './models'
  * @param {ApiOptions} options The API options object
  */
 export class Textile {
-  opts: ApiOptions
+  opts: ApiOptions = {
```
Comment (andrewxhill) [inline src/index.ts:31]: ApiOptions is now an internal method because it always requires a url, port and version

File: src/index.ts [hunk 2]
```diff
@@ -29,56 +29,62 @@ export * from './models'
...
   /** @property {Threads} threads - Manage Textile Threads */
-  threads: Threads
+  threads: Threads = new Threads(this.opts)
   /** @property {Tokens} tokens - Manage Textile Threads */
-  tokens: Tokens
+  tokens: Tokens = new Tokens(this.opts)
   /** @property {Utils} utils - Get information about the Textile node */
-  utils: Utils
-  constructor(options?: ApiOptions) {
+  utils: Utils = new Utils(this.opts)
```
Comment (andrewxhill) [inline src/index.ts:77]: no more constructor needed

File: src/index.ts [hunk 3]
```diff
@@ -29,56 +29,62 @@ export * from './models'
...
+
+/**
+ * Textile is the main client class
+ *
+ * @param {TextileOptions} options The Textile options object
+ */
+  setOptions(options?: TextileOptions) {
```
Comment (andrewxhill) [inline src/index.ts:84]: setOptions takes our new type, TextileOptions where port, url, version are all optionals and then just merges with the defaults

**Review by carsonfarmer — APPROVED** [review]
do it!

**Pull request merged**